daemon/algod/api/server/v2/handlers.go

```diff
@@ -259,15 +260,24 @@ func (v2 *Handlers) AccountInformation(ctx echo.Context, address string, params
 		return badRequest(ctx, err, errFailedToParseAddress, v2.Log)
 	}
 
+	// should we skip fetching apps and assets?
+	if params.Include != nil && *params.Include == "none" {
+		return v2.basicAccountInformation(ctx, addr, handle, contentType)
+	}
+
 	myLedger := v2.Node.Ledger()
-	record, lastRound, err := myLedger.LookupLatest(addr)
+	record, lastRound, amountWithoutPendingRewards, err := myLedger.LookupLatest(addr)
 	if err != nil {
 		return internalError(ctx, err, errFailedLookingUpLedger, v2.Log)
 	}
 
-	consensus, err := myLedger.ConsensusParams(lastRound)
-	if err != nil {
-		return internalError(ctx, err, fmt.Sprintf("could not retrieve consensus information for last round (%d)", lastRound), v2.Log)
+	// check against configured total limit on assets/apps
+	if maxResults := v2.Node.Config().MaxAccountsAPIResults; maxResults != 0 {
```

Comment:
the idea here is correct, but you'd want to do this without retrieving all the assets from the ledger.
i.e. this would need to be broken into 
- get base account data : then, figure out if the account has too many resources. if so, fail.
- otherwise, go get the resources and return them.

Reply:
Yes, I was thinking of modifying LookupLatest to either take a limit parameter, or call LookupWithoutRewards first to check the limits, and then call LookupLatest again  if it's below the limit. I was thinking the second.

Reply:
Yes, the second is good.

Reply:
updated